unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ChatController/ChatController.cs

```diff
@@ -135,6 +137,60 @@ public void UpdateChannelMembers(string payload)
             OnUpdateChannelMembers?.Invoke(msg.channelId, msg.members);
         }
 
+    // called by kernel
+    [PublicAPI]
+    public void JoinChannelConfirmation(string payload)
+    {
+        var msg = JsonUtility.FromJson<ChannelInfoPayloads>(payload);
+
+        if (msg.channelInfoPayload.Length == 0)
+            return;
+
+        var channelInfo = msg.channelInfoPayload[0];
+        var channel = new Channel(channelInfo.channelId, channelInfo.name, channelInfo.unseenMessages, channelInfo.memberCount, channelInfo.joined, channelInfo.muted, channelInfo.description);
+        var channelId = channel.ChannelId;
+        
+        if (channels.ContainsKey(channelId))
+            channels[channelId].CopyFrom(channel);
+        else
+            channels[channelId] = channel;
+        
+        if(!autoJoinChannelList.Get().Contains(channelId))
```

Comment:
i was thinking a different approach, let me know what do you think.
- `IChatController` add a method: `JoinAutomaticChannel(string channelId)`
- `WorldChatWindowController` method `ConnectToAutoJoinChannels` call `chatControllerJoinAutomaticChannel(channelId)`
- `ChatController.JoinAutomaticChannel(string channelId)` should flag the channels that are expected to be joined as automatic:
```
HashSet<string> requestedAutomaticChannelsToJoin; 
public void JoinAutomaticChannel(string channelId)
{
    requestedAutomaticChannelsToJoin.Add(channelId)
    JoinOrCreateChannel(channelId);
}
```
- when `JoinChannelConfirmation` arises:
```
public void JoinChannelConfirmation(string payload)
{
    ...
    var channel = solveChannelObject;
    if (requestedAutomaticChannelsToJoin.Contains(channel.ChannelId))
    {
        OnAutomaticChannelJoined?.Invoke(channel);
        requestedAutomaticChannelsToJoin.Remove(channel.ChannelId);
    }
    else
        OnChannelJoined?.Invoke(channel);

        OnChannelUpdated?.Invoke(channel);

    ... more stuff
}
```

Reply:
We could change it like this but I'd say to do it in a second stage when we will need to rework this functionality to add more things

Reply:
can you leave a TODO comment?